src/main/scala/org/ergoplatform/mining/difficulty/DifficultyAdjustment.scala

```diff
@@ -33,8 +32,56 @@ class LinearDifficultyControl(val chainSettings: ChainSettings) extends ScorexLo
     }
   }
 
+  private def bitcoinCalculate(previousHeaders: Seq[Header], epochLength: Int): Difficulty = {
+    val hs = previousHeaders.takeRight(2)
+    bitcoinCalculate(hs(0), hs(1), epochLength: Int)
+  }
+
+  /**
+    * Calculate difficulty as done in Bitcoin (with no capping result)
+    *
+    * Please note this method does not normalize its result!
+    *
+    * @param start - last block of previous epoch
+    * @param end - last block of current epoch
+    */
+  private def bitcoinCalculate(start: Header, end: Header, epochLength: Int): BigInt = {
+    end.requiredDifficulty * desiredInterval.toMillis * epochLength / (end.timestamp - start.timestamp)
+  }
+
+  /**
+    * Calculate difficulty for first block of a new epoch according to EIP-37
+    * @param previousHeaders - last headers of few epochs (8 in case of Ergo mainnet)
+    * @param epochLength - epoch length
+    */
+  def eip37Calculate(previousHeaders: Seq[Header], epochLength: Int): Difficulty = {
+    require(previousHeaders.size >= 2, "at least two headers needed for diff recalc")
+    val lastDiff = previousHeaders.last.requiredDifficulty
+
+    val predictiveDiff = calculate(previousHeaders, epochLength)
+    val limitedPredictiveDiff = if (predictiveDiff > lastDiff) {
+      predictiveDiff.min(lastDiff * 3 / 2)
+    } else {
+      predictiveDiff.max(lastDiff * 2 / 3)
```

Comment:
`lastDiff * 2 / 3` is 66.66%. 50% should be `lastDiff / 2`. 
The same goes for `avg`.

Reply:
I think the intention here is to make these adjustments mutually inverse.

Reply:
Yes, that was the intention, changed to / 2 though.